app/data_model/relationship_store.py

```diff
@@ -0,0 +1,95 @@
+from dataclasses import dataclass, asdict
+from typing import List, Mapping
+
+
+@dataclass
+class Relationship:
+    """
+    Represents a relationship between two items.
+    """
+
+    from_list_item_id: str
+    to_list_item_id: str
+    relationship: str
+
+    def for_json(self) -> Mapping:
+        return asdict(self)
+
+
+class RelationshipStore:
+    """
+    Stores and updates relationships.
+    """
+
+    def __init__(self, relationships: List[Mapping] = None) -> None:
+        self._is_dirty = False
+        self._relationships = self._build_map(relationships or [])
+
+    def __iter__(self):
+        return iter(self._relationships.values())
+
+    def __contains__(self, relationship):
+        return (
+            relationship.from_list_item_id,
+            relationship.to_list_item_id,
+        ) in self._relationships
+
+    def __len__(self):
+        return len(self._relationships)
+
+    @property
+    def is_dirty(self):
+        return self._is_dirty
+
+    def clear(self):
+        self._relationships.clear()
+        self._is_dirty = True
+
+    def serialise(self):
+        return [
+            relationship.for_json() for relationship in self._relationships.values()
+        ]
+
+    def get_relationship(self, from_list_item_id, to_list_item_id):
+        key = (from_list_item_id, to_list_item_id)
+        return self._relationships.get(key)
+
+    def add_or_update(self, relationship: Relationship):
+        key = (relationship.from_list_item_id, relationship.to_list_item_id)
+
+        existing_relationship = self._relationships.get(key)
+
+        if existing_relationship != relationship:
+            self._is_dirty = True
+
+        self._relationships[key] = relationship
+
+    def remove_all_relationships_for_list_item_id(self, list_item_id: str):
```

Comment:
The card mentioned that removing a person isn't handled but we have this method to do it, so we can use it here?
https://github.com/ONSdigital/eq-survey-runner/blob/c62316b70e3200192f6da5a99949f5d4ab304237/app/views/questionnaire.py#L477-L479
Were there other caveats around it hence we chose to not use it?

Reply:
It would add relationship specific code to the list collector views. I think we need to think about this and potentially have a solution where the list collector actions (add/edit/remove) can be monitored by observers (e.g. relationships) and acted on.

Reply:
Also there would be conflicts with the primary person collector work currently in progress.